### PR TITLE
Bump CUDA from 11.2 to 11.8

### PR DESCRIPTION
**Pull request by weiji14**
The CUDA 11.8 migration across conda-forge is practically complete (see https://conda-forge.org/status/#cuda118), so we can start updating to a newer version of CUDA!

This should add support for the new NVIDIA Ada Lovelace and [Hopper](https://developer.nvidia.com/blog/nvidia-hopper-architecture-in-depth/) generation GPUs that requires [compute capability](https://developer.nvidia.com/cuda-gpus) 8.9 or 9.0 (see https://docs.nvidia.com/cuda/archive/11.8.0/hopper-compatibility-guide/index.html#verifying-hopper-compatibility-using-cuda-11-8).

Note:
- There are actually CUDA 12.0 builds on conda-forge already, but it's probably good to have some docker images with CUDA 11.8 first, and transition to CUDA 12 later.
- CUDA 11.x has good forward and backwards compatibilty (see https://docs.nvidia.com/deploy/cuda-compatibility/index.html#cuda-intro), and as long as folks are using CUDA driver 450.36.06+, it should be ok.

Changes in this PR:
- [x] Update Pytorch, Torchvision and Tensorflow to use CUDA 11.8 builds
- [x] Update minimum pin on Tensorflow from >=2.9.1 to >=2.14.0 because lower versions <2.13.1 only has CUDA 11.2 on conda-forge.

References:
- https://developer.nvidia.com/blog/cuda-toolkit-11-8-new-features-revealed/
- https://github.com/conda-forge/conda-forge.github.io/issues/1981
- https://github.com/conda-forge/conda-forge-pinning-feedstock/pull/4834

### CODE REVIEW

File: pytorch-notebook/environment.yml [hunk 1]
```diff
@@ -7,6 +7,6 @@ channels:
 dependencies:
  - jupyterlab-nvdashboard
  - gpytorch
- - pytorch>=2.0.0=*cuda112*
- - torchvision>=0.15.1=*cuda112*
+ - pytorch>=2.0.0=*cuda118*
```
Comment (scottyhq) [inline pytorch-notebook/environment.yml:10]: Apparently we've been on 11.8 for some time, so perhaps this * syntax isn't really pinning anything? https://github.com/pangeo-data/pangeo-docker-images/blob/d2651fffe3f564eda66f873c8d052254b9f6d6de/pytorch-notebook/packages.txt#L114-L115

Comment (weiji14) [reply pytorch-notebook/environment.yml:10]: That's cudatoolkit. CUDA 11.2 and 11.8 does have this forward/backward compatibility thing, but if you look at the conda-lock.yml file, the Pytorch build is actually the CUDA 11.2 one:

https://github.com/pangeo-data/pangeo-docker-images/blob/d2651fffe3f564eda66f873c8d052254b9f6d6de/pytorch-notebook/conda-lock.yml#L9696

**Comment by scottyhq** [comment]
looks like tensorflow image size dropped a bit, but pytorch keeps inflating :) would be nice to get these below 10GB if possible one of these days

```
pangeo/ml-notebook 
10.8GB ->  10GB

pangeo/pytorch-notebook   
13.7GB -> 13.9GB
```

**Review by scottyhq — APPROVED** [review]

**Comment by weiji14** [comment]
Yes, things should get smaller! This is because conda-forge has removed the need for a large `cudatoolkit` package in CUDA 12 (see https://github.com/conda-forge/conda-forge.github.io/issues/1963), breaking it into smaller components that are installed as needed. So hopefully that can cut down a few hundred megabytes :crossed_fingers:

**Pull request merged**